### PR TITLE
Add global dependency validation pipeline task

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change adds the validateassets.ps1 script and relevant pipeline task to enable automated  validation of asset dependencies. This was identified as important due to the discovery of incorrect file dependencies in packages with the release of 2.5.0

#8944 tracks expanding the script to support scoped validation of only changed asset files.

### CODE REVIEW

File: pipelines/templates/tasks/validateassets.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  # associated with a given pull request. Note that this is only valid to use
+  # for pull request validation, since this functionality requires a pull request
+  # ID in order to determine the two commits to diff. Only relevant if validateCode
+  # is true.
+  scopedValidation: false
+  # The file containing the list of changes, used if scopedValidation is true.  
+  changesFile: ''
+
+steps:
+#v- ${{ if eq(parameters.scopedValidation, false) }}:
```
Comment (keveleigh) [inline pipelines/templates/tasks/validateassets.yml:12]: Should this be removed if it's commented out?
Also the comment syntax seems to have an extra `v`

File: scripts/ci/validateassets.ps1 [hunk 2]
```diff
@@ -0,0 +1,442 @@
+<#
```
Comment (keveleigh) [inline scripts/ci/validateassets.ps1:1]: missing MSFT license header

File: pipelines/templates/tasks/validateassets.yml (hunk 1 above)
Comment (davidkline-ms) [reply pipelines/templates/tasks/validateassets.yml:12]: it is commented temporarily, pending work being done as part of #8944

Comment (davidkline-ms) [reply pipelines/templates/tasks/validateassets.yml:12]: will fix the extra 'v'

File: scripts/ci/validateassets.ps1 (hunk 2 above)
Comment (davidkline-ms) [reply scripts/ci/validateassets.ps1:4]: added

File: pipelines/templates/tasks/validateassets.yml (hunk 1 above)
Comment (davidkline-ms) [reply pipelines/templates/tasks/validateassets.yml]: fixed the 'v', added comment re: #8944

**Review by MaxWang-MS — APPROVED** [review]
Looks good to me!

File: scripts/ci/validateassets.ps1 [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+        }
+
+        $numInvalid
+    }
+}
+
+# ####################
+# Start of main script
+# ####################
+GatherFileGuids
```
[the remark this answers is not in the document]
Comment (davidkline-ms) [reply scripts/ci/validateassets.ps1:424]: i was hitting trouble returning a hashtable.

happy to try again

[the remark this answers is not in the document]
Comment (davidkline-ms) [reply scripts/ci/validateassets.ps1:425]: looks like i was just doing something wrong before.... updating now

[the remark this answers is not in the document]
Comment (davidkline-ms) [reply scripts/ci/validateassets.ps1]: fixed

**Pull request merged**